pinot-server/src/main/java/org/apache/pinot/server/starter/helix/FreshnessBasedConsumptionStatusChecker.java

```diff
@@ -37,9 +40,15 @@ public class FreshnessBasedConsumptionStatusChecker extends IngestionBasedConsum
   private final long _minFreshnessMs;
   private final long _idleTimeoutMs;
 
-  public FreshnessBasedConsumptionStatusChecker(InstanceDataManager instanceDataManager, Set<String> consumingSegments,
+  public FreshnessBasedConsumptionStatusChecker(InstanceDataManager instanceDataManager,
+      Map<String, Set<String>> consumingSegments, long minFreshnessMs, long idleTimeoutMs) {
+    this(instanceDataManager, consumingSegments, null, minFreshnessMs, idleTimeoutMs);
+  }
+
+  public FreshnessBasedConsumptionStatusChecker(InstanceDataManager instanceDataManager,
```

Comment:
Since we are modifying the API anyway, maybe we should enforce a non-null supplier?

Reply:
sgtm